### PR TITLE
XML: comment and PI in doctype internal subset

### PR DESCRIPTION
**Pull request by zcorpan**
Per the XML Information Set spec, comments in the doctype are not represented and PIs are children of the doctype. The DOM does not allow children of DocumentType nodes. Therefore, in the DOM, they should not be represented.

### CODE REVIEW

**Review by annevk — APPROVED** [review]
I think it would be better if there are no references to the infoset, which I think we ought to largely ignore in favor of the combination of the HTML Standard, DOM Standard, and XML + XMLNS standards.

File: css/cssom/xml-stylesheet-pi-in-doctype.xhtml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html [<?xml-stylesheet href="data:text/css,html{z-index: 1}"?>]>
+<html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>xml-stylesheet processing instruction in doctype internal subset</title>
+    <link rel="help" href="https://www.w3.org/TR/2004/REC-xml-infoset-20040204/#infoitem.pi"/>
+    <!-- "[parent] The document, element, or document type declaration information item
+    which contains this information item in its [children] property. " -->
```
Comment (zcorpan) [inline css/cssom/xml-stylesheet-pi-in-doctype.xhtml:7]:
```suggestion
```

File: dom/nodes/Comment-in-doctype.xhtml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html [<!--x-->]><html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>XML: Comment in doctype internal subset</title>
+    <link rel="help" href="https://www.w3.org/TR/2004/REC-xml-infoset-20040204/#infoitem.comment"/>
+    <!-- "There is a comment information item for each XML comment in the original document,
+    except for those appearing in the DTD (which are not represented)." -->
```
Comment (zcorpan) [inline dom/nodes/Comment-in-doctype.xhtml:6]:
```suggestion
```

File: dom/nodes/ProcessingInstruction-in-doctype.xhtml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html [<?x y?>]><html xmlns="http://www.w3.org/1999/xhtml">
+  <head>
+    <title>XML: Processing instruction in doctype internal subset</title>
+    <link rel="help" href="https://www.w3.org/TR/2004/REC-xml-infoset-20040204/#infoitem.pi"/>
+    <!-- "[parent] The document, element, or document type declaration information item
+    which contains this information item in its [children] property. " -->
```
Comment (zcorpan) [inline dom/nodes/ProcessingInstruction-in-doctype.xhtml:6]:
```suggestion
```

**Pull request merged**